logger.go

```diff
@@ -178,11 +180,57 @@ func (log *Logger) With(fields ...Field) *Logger {
 	if len(fields) == 0 {
 		return log
 	}
+	if log.deferWith {
+		log.core = &deferredWithCore{Core: log.core, fields: fields}
```

Comment:
this will cause cross talk
```suggestion
                 l := *log
		l.core = &deferredWithCore{Core: log.core, fields: fields}
		return &l
```